limitador/src/storage/mod.rs

```diff
@@ -140,19 +140,18 @@ impl AsyncStorage {
         self.limits.read().unwrap().keys().cloned().collect()
     }
 
-    pub fn add_limit(&self, limit: Limit) {
+    pub fn add_limit(&self, limit: Limit) -> bool {
```

Comment:
> Also, it interesting to account for what it'd mean when RateLimitPolicy are compiled down to the single Limitador CR… what can we already see there? what should make it to the ConfigMap to eventually be picked up by the Limitador server and possibly reflect it thru some (REST?) API to the individual RateLimitPolicy CRs…

I am not sure I follow. You mean add some info in the status of the RLP?  FYI, check the schema of the [RLP new design](https://github.com/Kuadrant/kuadrant-controller/blob/main/doc/proposals/rlp-target-gateway-resource.md#schema-crd-of-the-ratelimitpolicy), 

```yaml
 limits:
        - conditions: ["admin == yes"]
          max_value: 500
          seconds: 30
          variables: []
```
no namespace (aka domain) is defined. It is up to the control plane to assign namespace to it.

Reply:
What if two people define two "equivalent" `RateLimitPolicy` CRs (logically speaking, so say they have different `max_value`, but possibly the same, it doesn't really matter _I think_), when compiling the `Limitador` CR with all the `Limit`s from the different `RateLimitPolicy` CRs, we might have a conflict now… 

What does that mean? Who (the operator, Limitador, ...?) makes what call? How is it then reflected on the individual CRs' status? How does that affect further changes (valid new `RateLimitPolicy` CRs being created, or existing changed while the conflicting ones are still there)? If limitador rejects a _complete_ new Limit file, these question matter.

Ideally, the operator would be able to "ignore" these CRs and compile to an always valid limitador limit file. But is this even possible? Do we need to duplicate logic? Expose some service to test a config before applying it?

Reply:
Good questions here. 

The kuadrant control plane, the owner of the content in the Limitador CR in the kuadrant context, should do the validations. And if any conflict is found, reject the RLP adding the reason in the status field.

The limitador operator can also run some validation checks to avoid such invalid limit definitions. 

I will think about that and add it to the [proposal doc](https://github.com/Kuadrant/kuadrant-controller/blob/main/doc/proposals/rlp-target-gateway-resource.md). This information should end up in the kuadrant's doc about the behavior of the control plane.